src/duplicacy_b2client.go

```diff
@@ -131,11 +140,11 @@ func (client *B2Client) call(url string, input interface{}) (io.ReadCloser, int6
 				backoff = client.retry(backoff, response)
 				continue
 			}
-			return nil, 0, err
+			return nil, nil, 0, err
 		}
 
 		if response.StatusCode < 300 {
-			return response.Body, response.ContentLength, nil
+			return response.Body, response.Header, response.ContentLength, nil
 		}
 
```

Comment:
I think you need to check if the response code is 404 and the method is HEAD.  If so return nil httpHeader.  Without this check it will keep retrying when looking up a non-existent file.